server/call/session.py

```diff
@@ -348,18 +365,23 @@ def on_participant_joined(self, participant):
         # As soon as someone joins, stop shutdown process if one is in progress
         if self._shutdown_timer:
             self._logger.info("Participant joined - cancelling shutdown.")
-            self._shutdown_timer.cancel()
-            self._shutdown_timer = None
+            self.cancel_shutdown_timer()
 
     def on_participant_left(self,
                             participant,
                             reason):
         """Callback invoked when a participant leaves the Daily room."""
         self.maybe_start_shutdown()
 
+    def on_call_state_updated(self, state: Mapping[str, Any]) -> None:
+        self._logger.info("Call state updated for session %s: %s", self._id, state)
+        if state == "left" and not self._is_destroyed:
+            self._logger.info("Call state left, destroying immediately")
+            self.on_left_meeting(None)
+
     def maybe_start_shutdown(self) -> bool:
         """Checks if the session should be shut down, and if so, starts the shutdown process."""
-        count = self._call_client.participant_counts()['present']
+        count = self.get_participant_count()
```

Comment:
For now using REST API for this because we saw some odd/inaccurate participant leave/join event logs from the python client that seemed like they may be impacting the participant counts returned.